graphbolt/src/csc_sampling_graph.cc

```diff
@@ -122,9 +122,12 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
 }
 
 c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
-    const torch::Tensor& nodes, int64_t fanout, bool replace) const {
+    const torch::Tensor& nodes, const std::vector<int64_t>& fanouts,
+    bool replace, const torch::optional<torch::Tensor>& probs) const {
   const int64_t num_nodes = nodes.size(0);
-
+  // If true, perform sampling for each edge type of each node, otherwise just
+  // sample once for each node with no regard of edge types.
+  bool consider_etype = (fanouts.size() > 1);
```

Comment:
const

Reply:
Do we check `fanouts.size() == num_etypes` for this case?

Reply:
Add this logic.